### ISSUE TITLE
router-bgp enhancements and modifications to make eos_config_cli_gen deployment model agnostic

### ISSUE THREAD

**Issue by MilesTails** [aristanetworks/ansible-avd#323]
**Issue Type**

- Role enhancement

**Is your feature request related to a problem? Please describe.**
After reviewing the BGP VRF sub-mode capabilities within eos_cli_config_gen I can see that the parameters have been restricted to that of a EVPN type configuration. I.e RD/Route Targets are mandatory. 

**Describe the solution you'd like**
- Change EVPN parameters validation check from mandatory to optional.
- Addition of local-as and route-reflector-client options within all appropriate sections of the BGP model.
- Addition of address families within vrf sub-mode.

**Describe alternatives you've considered**
Considered the idea of a flag to denote EVPN VRF deployment however my thoughts are that these restrictions should be handled in the template that is generating the structure to feed into eos_cli_config_gen. eos_cli_config_gen should be deployment model agnostic.

**Additional context**
Proposed enhancements and data model modifications:
``` yaml
router_bgp:
  neighbors:
    < IPv4_address_1 >:
      route_reflector_client: < true | false >
  peer_groups:
    < peer_group_name_1>:
      route_reflector_client: < true | false > # ADD TO DOC
  vrfs:
    < vrf_name_1 >:
      neighbors:
        < neighbor_ip_address >:
          shutdown: < true | false >
          bfd: < true | false >
          weight: < weight_value > # ADD TO DOC
          route_reflector_client: < true | false >
      address_family_ipv4:
        peer_groups:
          < peer_group_name >:
            route_map_in: < route_map_name >
            route_map_out: < route_map_name >
            activate: < true | false >
            prefix_list_in: < prefix_list_name >
            prefix_list_out: < prefix_list_name >
        neighbors:
          < neighbor_ip_address>:
            prefix_list_in: < prefix_list_name >
            prefix_list_out: < prefix_list_name >
            default_originate:
              always: < true | false >
              route_map: < route_map_name >
      address_family_ipv4_multicast:
        peer_groups:
          < peer_group_name >:
            activate: < true | false >
        redistribute_routes:
          < route_type >:
      address_family_ipv6:
        peer_groups:
          < peer_group_name >:
            activate: < true | false >
            route_map_in: < route_map_name >
            route_map_out: < route_map_name >
```

**Comment by titom73** [comment]
Might be linked to #128 as we might need to implement new BGP knobs in `eos_l3ls_evpn`

**Comment by ClausHolbechArista** [comment]
Most of this has been implemented in #2298, #2808 and others. Please reevaluate if any of this is still relevant. Closing as stale.
